src/native/managed/cdacreader/src/Contracts/Metadata_1.cs

```diff
@@ -0,0 +1,333 @@
+    {
+        return new NonValidatedMethodTable_1(_target, methodTablePointer);
+    }
+
+    private NonValidatedEEClass_1 GetUntrustedEEClassData(TargetPointer eeClassPointer)
+    {
+        return new NonValidatedEEClass_1(_target, eeClassPointer);
+    }
+
+    public MethodTableHandle GetMethodTableHandle(TargetPointer methodTablePointer)
+    {
+        // if we already validated this address, return a handle
+        if (_methodTables.ContainsKey(methodTablePointer))
+        {
+            return new MethodTableHandle(methodTablePointer);
+        }
+        // Check if we cached the underlying data already
+        if (_target.ProcessedData.TryGet(methodTablePointer, out Data.MethodTable? methodTableData))
+        {
+            // we already cached the data, we must have validated the address, create the representation struct for our use
+            MethodTable_1 trustedMethodTable = new MethodTable_1(methodTableData);
+            _ = _methodTables.TryAdd(methodTablePointer, trustedMethodTable);
+            return new MethodTableHandle(methodTablePointer);
+        }
+
+        // Otherwse, get ready to validate
+        NonValidatedMethodTable_1 nonvalidatedMethodTable = GetUntrustedMethodTableData(methodTablePointer);
+
+        // if it's the free object method table, we trust it to be valid
+        if (methodTablePointer == FreeObjectMethodTablePointer)
+        {
+            Data.MethodTable freeObjectMethodTableData = _target.ProcessedData.GetOrAdd<Data.MethodTable>(methodTablePointer);
+            MethodTable_1 trustedMethodTable = new MethodTable_1(freeObjectMethodTableData);
+            _ = _methodTables.TryAdd(methodTablePointer, trustedMethodTable);
+            return new MethodTableHandle(methodTablePointer);
+        }
+        if (!ValidateMethodTablePointer(nonvalidatedMethodTable))
+        {
+            throw new ArgumentException("Invalid method table pointer");
+        }
+        // ok, we validated it, cache the data and add the MethodTable_1 struct to the dictionary
+        Data.MethodTable trustedMethodTableData = _target.ProcessedData.GetOrAdd<Data.MethodTable>(methodTablePointer);
+        MethodTable_1 trustedMethodTableF = new MethodTable_1(trustedMethodTableData);
+        _ = _methodTables.TryAdd(methodTablePointer, trustedMethodTableF);
+        return new MethodTableHandle(methodTablePointer);
+    }
+
+    private bool ValidateMethodTablePointer(NonValidatedMethodTable_1 umt)
+    {
+        try
+        {
+            if (!ValidateWithPossibleAV(umt))
+            {
+                return false;
+            }
+            if (!ValidateMethodTable(umt))
+            {
+                return false;
+            }
+        }
+        catch (System.Exception)
+        {
+            // FIXME: maybe don't swallow all exceptions?
+            return false;
+        }
+        return true;
+    }
+
+    private bool ValidateWithPossibleAV(NonValidatedMethodTable_1 methodTable)
```

Comment:
Is this saying it will A/V on the debuggee side or debugger? I assume the debuggee side since we just call it and we can't catch an A/V in managed code any longer. Perhaps it would be more accurate to simple say `ValidateMethodTableUnsafe()` and document the potential consequences instead of encoding that in the name.

Reply:
Can I assume this will not be an `AccessViolationException`? If so, then yes please update to `ValidateThrowing()`.